client/landing/gutenboarding/components/domain-picker/index.tsx

```diff
@@ -120,6 +120,24 @@ const DomainPicker: FunctionComponent< Props > = ( {
 		if ( allSuggestions ) {
 			setRailcarId( getNewRailcarId() );
 		}
+
+		if ( freeSuggestions && recommendedSuggestion && paidSuggestions ) {
```

Comment:
I haven't actually tested whether I need this check. My linter was complaining so this shut it up.

Reply:
Yes if `allSuggestions` is `undefined`, then these will be `undefined` too. And that will happen when the search term is empty.

If you wanted you could remove a level of nesting maybe by only testing for truthyness on the branches where `freeSuggestions` etc. are actually used? But this is fine.